akka-stream/src/main/scala/akka/stream/ActorMaterializer.scala

```diff
@@ -129,12 +129,11 @@ object ActorMaterializer {
    * INTERNAL API
    */
   private[akka] def downcast(materializer: Materializer): ActorMaterializer =
-    materializer match {
+    materializer match { //FIXME this method is going to cause trouble for other Materializer implementations
```

Comment:
If someone would try to materialize a SynchronousFileSource for instance, with a non ActorMaterializer, it would simply fail. And there'd be no way for them to write their Materializer so that it wouldn't fail.